designs/0019-variadic_odes.md

```diff
@@ -0,0 +1,171 @@
+- Feature Name: Variadic arguments for ODE solvers
+- Start Date: 2020-05-29
+- RFC PR: 19
+- Stan Issue:
+
+# Summary
+[summary]: #summary
+
+This proposal is to make a set of ODE solvers that take a variadic list of
+arguments.
+
+# Motivation
+[motivation]: #motivation
+
+The goal is to avoid the argument packing and unpacking that the current solvers
+depend on.
+
+Argument packing is inconvenient, makes code hard to read, and introduces more
+chances for making errors.
```

Comment:
Maybe add as advantage that the requirement for packing leads easily to over-propagation of packed data to properly named variables which are cast to `vars` which is avoided with the new variadic interface.

Reply:
Can't conceptually we put all data in the x_r arguments so there's no promotion?